File: src/turnopos.client/src/models/Item.ts
import type Department from "./Department";

export default interface Item {
    id: number;
    isDirectory: boolean;
    parentId?: number | null;
    departmentId?: number | null;
    name: string;
    description?: string;
    price?: number;
    stock?: number;
    order?: number;
    isActive: boolean;
    parent?: Item | null;
    department?: Department | null;
    children?: Item[];
    childrenLoaded: boolean;
}

File: src/TurnoPOS.Data/Models/Item.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using TurnoPOS.Data.Models.Base;

namespace TurnoPOS.Data.Models;

public class Item : BaseEntity
{
    public bool IsDirectory { get; set; } = false;
    public long? ParentId { get; set; }
    public long? DepartmentId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int Stock { get; set; }
    public int Order { get; set; }
    public bool IsActive { get; set; } = true;

    [ForeignKey(nameof(ParentId)), JsonIgnore]
    public Item? Parent { get; set; }

    [ForeignKey(nameof(DepartmentId))]
    public Department? Department { get; set; }

    public ICollection<Item> Children { get; set; } = [];
}
